spark/src/test/java/org/apache/iceberg/spark/data/TestSparkOrcReader.java

```diff
@@ -71,7 +71,7 @@ private void writeAndValidateRecords(Schema schema, Iterable<InternalRow> expect
 
     try (CloseableIterable<InternalRow> reader = ORC.read(Files.localInput(testFile))
         .project(schema)
-        .createReaderFunc(SparkOrcReader::new)
```

Comment:
I think this should test with and without container reuse if that is implemented in this PR. Probably just make this test parameterized.

Reply:
For now I've removed the reuse code. We can tackle than in followup